### PR TITLE
Make `stake-address-info` query also return deposit

### LINKED ISSUE #353: [FEATUE] A query for the stake key deposited amount | Stake-Address-Info is not showing the deposited amount
Version Info:
```
cardano-cli 8.12.0.0 - linux-x86_64 - ghc-8.10
git rev f1ce770834bf7150ca29cb647065c9e62d39be1a
```
**PROBLEM STATEMENT:** 

With the new stake address registration certificate type7 in conway era, we can now set a deposit amount. That exact deposit amount should also be used to generate the de-registration/retire certificate. For that we would need a method to query the registered amount. Best place imo would be within the stake-address-info query.

Currently we get something like:
```json
[
  {
    "address": "stake_test1urqntq4wexjylnrdnp97qq79qkxxvrsa9lcnwr7ckjd6w0cr04y4p",
    "delegation": "pool1rkfs9glmfva3jd0q9vnlqvuhnrflpzj4l07u6sayfx5k7d788us",
    "rewardAccountBalance": 297059772555
  }
]
```

So adding a new entry `deposited` would be a good solution imo to query the correct value before generating the de-registration/retire certificate.

P.S.: Same goes for the dRep information query. That should also list the deposited amount if possible.

### PR DESCRIPTION
**Pull request by carlhammann**
# Changelog

```yaml
- description: |
    `cardano-cli query stake-address-info` now returns the deposit for the currently delegated staking credential (in addition to what it returned before).
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This PR closes #353.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

### CODE REVIEW

**Review by Jimbo4350 — APPROVED** [review]
I like the improvement but can you inform QA about this change. They may depend on the output.

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs [hunk 1]
```diff
@@ -902,12 +902,19 @@ runQueryStakeAddressInfoCmd
...
           & onLeft (left . QueryCmdLocalStateQueryError . EraMismatchError)
 
-        pure $ do
-          writeStakeAddressInfo mOutFile $ DelegationsAndRewards result
+        stakeDelegDeposits <- lift (queryStakeDelegDeposits eInMode sbe stakeAddr)
+          & onLeft (left . QueryCmdUnsupportedNtcVersion)
+          & onLeft (left . QueryCmdLocalStateQueryError . EraMismatchError)
+
+        return $ do
+          writeStakeAddressInfo
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs:914]: Is it worth updating the function name to say `writeStakeAddressInfoAndBalances`?

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs [hunk 2]
```diff
@@ -917,13 +924,42 @@ runQueryStakeAddressInfoCmd
...
-        $ LBS.writeFile fpath (encodePretty delegsAndRewards)
+writeStakeAddressInfo
+  mOutFile
+  (DelegationsAndRewards (stakeAccountBalances, stakePools))
+  stakeDelegDeposits =
+    case mOutFile of
+      Nothing -> liftIO $ LBS.putStrLn (encodePretty jsonInfo)
+      Just (File fpath) ->
+        handleIOExceptT (QueryCmdWriteFileError . FileIOError fpath)
+          $ LBS.writeFile fpath (encodePretty jsonInfo)
```
Comment (newhoggy) [inline cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs]: You should be able to use this function here? https://github.com/input-output-hk/cardano-api/blob/newhoggy/new-shelleyToAlonzoEraToShelleyToBabbageEra-function/cardano-api/internal/Cardano/Api/IO.hs/#L115-L123

**Review by newhoggy — APPROVED** [review]

File: cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs (hunk 1 above)
Comment (carlhammann) [reply cardano-cli/src/Cardano/CLI/EraBased/Run/Query.hs:914]: Not sure; the name you propose wouldn't even have been completely descriptive before this change, but now we write balances, stake pools, and deposits... In short, all information associated with a stake address that we can query.

**Pull request merged**